runtime/vm/BufferWriter.hpp

```diff
@@ -342,6 +342,19 @@ class VM_BufferWriter {
 		writeU8(val ? 1 : 0);
 	}
 
+	void
+	writeFormattedString(const char *format, ...)
+	{
+		va_list args;
+		va_start(args, format);
+		int totalLength = vsnprintf((char *)_cursor, _bufferEnd - _cursor, format, args);
+		va_end(args);
+
+		if (checkBounds(totalLength)) {
```

Comment:
you should check bounds before you write to the buffer

Reply:
Fixed

Reply:
The other use of `vsnprintf()` should be updated.